filecoin-proofs/src/api/post.rs

```
@@ -220,14 +223,32 @@ pub fn generate_candidates(
     let unique_trees_res: Vec<_> = unique_challenged_replicas
         .into_par_iter()
         .map(|(id, replica)| {
+            // Ensure that any associated cached data persisted is discarded and our tree is compacted by this point.
+            let f_aux_path = replica.cache_dir_path().join(CacheKey::TAux.to_string());
+            if Path::new(&f_aux_path).exists() {
```

this should probably return an error if the path does not exist